marlowe/src/Language/Marlowe/Client.hs

```diff
@@ -465,15 +472,50 @@ setupMarloweParams owners contract = mapError (review _MarloweError) $ do
                             <> Constraints.mustMintValue (Currency.mintedValue theCurrency)
         let rolesSymbol = Ledger.scriptCurrencySymbol curVali
         let minAdaTxOut = adaValueOf 2
-        let giveToParty (role, pkh) = Constraints.mustPayToPubKey pkh (Val.singleton rolesSymbol role 1 <> minAdaTxOut)
-        let distributeRoleTokens = foldMap giveToParty $ second PaymentPubKeyHash <$> AssocMap.toList owners
+        let giveToParty (role, addr) =
+              mustPayToShelleyAddress addr (Val.singleton rolesSymbol role 1 <> minAdaTxOut)
+        distributeRoleTokens <- foldMapM giveToParty $ AssocMap.toList owners
         let params = marloweParams rolesSymbol
         pure (params, mintTx <> distributeRoleTokens, lookups)
     else do
         let missingRoles = roles `Set.difference` Set.fromList (AssocMap.keys owners)
         let message = T.pack $ "You didn't specify owners of these roles: " <> show missingRoles
         throwing _ContractError $ OtherError message
 
+ownShelleyAddress
+  :: AddressInEra ShelleyEra
+  -> Contract MarloweContractState s MarloweError (ScriptLookups Void)
+ownShelleyAddress addr = Constraints.ownPaymentPubKeyHash . fst <$> shelleyAddressToKeys addr
+
+mustPayToShelleyAddress
+  :: AddressInEra ShelleyEra
+  -> Val.Value
+  -> Contract MarloweContractState s MarloweError (TxConstraints i o)
+mustPayToShelleyAddress addr value = do
+  (ppkh, skh) <- shelleyAddressToKeys addr
+  pure $ ($ value) $ maybe
+    (Constraints.mustPayToPubKey ppkh)
+    (Constraints.mustPayToPubKeyAddress ppkh)
+    skh
+
+shelleyAddressToKeys
+  :: AddressInEra ShelleyEra
+  -> Contract MarloweContractState s MarloweError (PaymentPubKeyHash, Maybe StakePubKeyHash)
+shelleyAddressToKeys (AddressInEra _ (Shelley.ShelleyAddress _ paymentCredential stakeRef)) =
+  case Shelley.fromShelleyPaymentCredential paymentCredential of
+    PaymentCredentialByScript _ -> throwError $ OtherContractError $ OtherError "Script payment addresses not supported"
+    PaymentCredentialByKey hash ->
+      let ppkh = PaymentPubKeyHash . PubKeyHash . toBuiltin $ serialiseToRawBytes hash
+      in
+        case Shelley.fromShelleyStakeReference stakeRef of
+          StakeAddressByValue (StakeCredentialByScript _) ->
+            throwError $ OtherContractError $ OtherError "Script stake addresses not supported"
+          StakeAddressByPointer _ ->
+            throwError $ OtherContractError $ OtherError "Pointer stake addresses not supported"
+          NoStakeAddress -> pure (ppkh, Nothing)
+          StakeAddressByValue (StakeCredentialByKey stakeHash) ->
+            pure (ppkh,  Just . StakePubKeyHash . PubKeyHash . toBuiltin $ serialiseToRawBytes stakeHash)
+shelleyAddressToKeys _ = throwError $ OtherContractError $ OtherError "Byron Addresses not supported"
```

Comment:
It actually wouldn't be difficult to support Byron addresses, the `Cardano.Api.Byron` module has address functions similar to `Cardano.Api.Shelley`.

Reply:
I'll merge this PR and we can add it if/when we need it